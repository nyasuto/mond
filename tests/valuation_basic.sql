-- prepare data
INSERT INTO assets (ticker, ccy) VALUES ('VTI','USD');
INSERT INTO fx_rates (date, pair, rate) VALUES ('2025-09-15','USDJPY',145.2);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2025-09-15','VTI',2037.88,270.5);

-- assert valuation (rounded for determinism)
SELECT
  date,
  ticker,
  ccy,
  qty,
  price_ccy,
  fx_rate,
  round(value_jpy, 3) AS value_jpy_3dp
FROM v_valuation
ORDER BY date, ticker;

