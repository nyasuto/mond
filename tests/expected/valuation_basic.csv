date,ticker,ccy,qty,price_ccy,fx_rate,value_jpy_3dp
2025-09-15,VTI,USD,2037.88,270.5,145.2,80040997.608
